src/App.tsx

```diff
@@ -3,33 +3,61 @@ import './App.scss';
 import { MoviesList } from './components/MoviesList';
 import moviesFromServer from './api/movies.json';
 
-type State = {};
+type State = {
+  query: string,
+};
 
 export class App extends React.Component<{}, State> {
-  state: State = {};
+  state: State = {
+    query: '',
+  };
+
+  handlerInput = (event: React.ChangeEvent<HTMLInputElement>) => {
+    this.setState({
+      query: event.target.value,
+    });
+  };
+
+  getVisibleMovies = (movies: Movie[]) => {
+    const { query } = this.state;
+
+    return movies.filter((value) => {
+      const title = value.title.toLowerCase();
+      const description = value.description.toLowerCase();
+      const lowerCaseQuery = query.toLowerCase();
```

Comment:
no need to do query to lowercase on every iteration, you can do it before filtering, because query doesn't change